src/main/java/com/peerrequest/app/api/DirectRequestsController.java

```diff
@@ -0,0 +1,264 @@
+                                                          @AuthenticationPrincipal OAuth2User user) {
+        if (this.entryService.get(entryId).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entry does not exist");
+        }
+
+        String researcherId = this.entryService.get(entryId).get().getResearcherId();
+        if (!user.getAttribute("sub").toString().equals(researcherId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the user that created the entry may create a request process");
+        }
+
+        if (dto.openSlots().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "open slots must be given");
+        }
+
+        if (dto.openSlots().get() < 0) {
+            throw new ResponseStatusException(HttpStatus.I_AM_A_TEAPOT, "open slots value must be greater than 0");
+        }
+
+        var option = this.directRequestProcessService.update(entryId, dto);
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "direct request process does not exist");
+        }
+
+        return option.get().toDto();
+    }
+
+    /**
+     * Gets a directRequests.
+     *
+     * @param requestId request to get
+     *
+     * @return directRequest with requestId
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests/{request_id}",
+            produces = "application/json")
+    public DirectRequest.Dto getDirectRequest(@PathVariable("entry_id") final Long entryId,
+                                              @PathVariable("request_id") final Long requestId,
+                                              @AuthenticationPrincipal OAuth2User user) {
+        if (this.entryService.get(entryId).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entry does not exist");
+        }
+
+        var option = this.directRequestService.get(requestId);
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "direct request does not exist");
+        }
+
+        String researcherId = this.entryService.get(entryId).get().getResearcherId();
+        String reviewerId = option.get().getReviewerId();
+        if (!user.getAttribute("sub").toString().equals(researcherId)
+                || user.getAttribute("sub").toString().equals(reviewerId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the researcher or reviewer may get the request");
+        }
+        return option.get().toDto();
+    }
+
+    /**
+     * Gets the list of all requests of an entry.
+     *
+     * @return directRequest with requestId
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests",
+            produces = "application/json")
```

Comment:
remove:  produce = "aplication/json"
to match the other controllers

Reply:
Done.